pandas/core/generic.py

```diff
@@ -6290,7 +6290,11 @@ def astype(
             for i, (col_name, col) in enumerate(self.items()):
                 cdt = dtype_ser.iat[i]
                 if isna(cdt):
-                    res_col = col.copy() if copy else col
+                    if using_copy_on_write():
+                        # Make a shallow copy even if copy=False for CoW
+                        res_col = col.copy(deep=copy)
+                    else:
+                        res_col = col if copy is False else col.copy()
```

Comment:
I think we can simplify this to just `res_col = col.copy(deep=copy)`. The resulting Series is still passed to `concat`, so it's not that we ever actually return this Series, so I would think that for the non-CoW case it shouldn't matter if the original column or a shallow copy of it is passed to concat?

Reply:
Yeah, true. Is a bit slower, but just a little bit :)